### PR TITLE
3.x: update rpc documentation

### PR DESCRIPTION
**Pull request by nathanfranke**
Fixes https://github.com/godotengine/godot-docs/issues/3917

Edit: To the mysterious person who approved the workflows, thanks :laughing:

### CODE REVIEW

File: tutorials/networking/high_level_multiplayer.rst [hunk 1]
```diff
@@ -235,23 +235,34 @@ You might have already noticed something different, which is the usage of the ``
...
+- ``mastersync``
+
+Each of them designate who can call the rpc, and optionally ``sync`` if the RPC can be called locally.
+
+.. note:: If no rpc keywords are added, Godot will block any attempts to call functions remotely.
+          This makes security work a lot easier (so a client can't call a function to delete a file on another client's system).
+
+The ``remote`` keyword can be called by any peer, including the server and all clients.
+The ``puppet`` keyword means a call can be made from the network master to any network puppet.
+The ``master`` keyword means a call can be made from any network puppet to the network master.
```
Comment (nathanfranke) [inline tutorials/networking/high_level_multiplayer.rst:254]: Even after using these keywords a ton, I still need someone to verify that I am describing them correctly here. Also I switched up the ordering to what is probably more likely to be used.

File: tutorials/networking/high_level_multiplayer.rst [hunk 2]
```diff
@@ -235,23 +235,34 @@ You might have already noticed something different, which is the usage of the ``
...
-The ``remote`` keyword means that the ``rpc()`` call will go via network and execute remotely.
+    puppetsync func update_position(new_position):
+        position = new_position
 
-The ``remotesync`` keyword means that the ``rpc()`` call will go via network and execute remotely, but will also execute locally (do a normal function call).
+.. tip:: You can also use :ref:`SceneTree.get_rpc_sender_id() <class_SceneTree_method_get_rpc_sender_id>` to have more advanced rules on how an rpc can be called.
 
-The others will be explained further down.
-Note that you could also use the ``get_rpc_sender_id`` function on ``SceneTree`` to check which peer actually made the RPC call to ``register_player``.
+These keywords are further explained in `Synchronizing the game`_.
```
Comment (nathanfranke) [inline tutorials/networking/high_level_multiplayer.rst:265]: Will this relative link work?

Comment (Faless) [reply tutorials/networking/high_level_multiplayer.rst:265]: I think you need to add a reference for it, so, here:
```
:ref:`Synchronizing the game <_doc_high_level_multiplayer_synchronizing>`
```

And where the title is:

```
.. _doc_high_level_multiplayer_synchronizing:

Synchronizing the game
----------------------
```

**Review by Faless — CHANGES_REQUESTED** [review]
The description looks good :+1: , see the comment about the reference.

**Review by Faless — APPROVED** [review]
Looking good to me :+1: .
Should be good to merge once CI passes.

**Comment by Faless** [comment]
I have no clue why the reference is not working :/... CC @Calinou ?

**Comment by nathanfranke** [comment]
I tried removing the underscore from the reference, since the SceneTree one also omitted it.

**Review by mhilbrunner — APPROVED** [review]
Build & tested locally, references work and the changes look good. Thank you!

**Comment by mhilbrunner** [comment]
Cherry-picked to 3.5.

**Pull request merged**